chore: add clarifying note for composite and expand term (#1078)
graphql/graphql-spec: spec/Section 5 -- Validation.md

- If {typeA} or {typeB} is Scalar or Enum:
  - If {typeA} and {typeB} are the same type return {true}, otherwise return
    {false}.

> - Assert: {typeA} and {typeB} are both Object, Interface or Union types.

Reviewer: This reads ambiguously; it could be interpretted as `(typeA is object and typeB is object) or (typeA is union and typeB is union) or (typeA is an interface and typeB is an interface)` rather than `(typeA is object or union or interface) and (typeB is object or union or interface)`. Perhaps:

```suggestion
- Assert: {typeA} is an object, union or interface type.
- Assert: {typeB} is an object, union or interface type.
```

  - If {SameResponseShape(subfieldA, subfieldB)} is {false}, return {false}.
- Return {true}.

> Note: composite is used to signal a type that is either an object-, interface-

Author: ```suggestion
Note: Composite is used to signal a type that is either an object-, interface-
```

> Note: In prior versions of the spec the term "composite" wass used to signal
> a type that is either an object-, interface- or union-type.

Reviewer: ```suggestion
Note: In prior versions of the spec the term "composite" was used to signal
a type that is either an Object, Interface or Union type.
```

}
```

> #### Fragments on object, interface or union Types

Reviewer: See the [style guide on titles](https://github.com/graphql/graphql-spec/blob/main/STYLE_GUIDE.md#title-case):

```suggestion
#### Fragments on Object, Interface or Union Types
```
